src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -484,7 +472,7 @@ private async Task<Compilation> GetOrBuildDeclarationCompilationAsync(SolutionSe
             /// Builds the compilation matching the project state. In the process of building, also
             /// produce in progress snapshots that can be accessed from other threads.
             /// </summary>
-            private Task<CompilationInfo> BuildCompilationInfoAsync(
+            private async Task<CompilationInfo> BuildCompilationInfoAsync(
```

Comment:
Did we need the async here?

Reply:
(we had gotten away with not needing the extra state machine before, not sure what forced it now)